cdap-docs/tools/docs-cli-commands.py

```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+#
+# Copyright © 2016 Cask Data, Inc.
+#
+# Used to generate a table in the CLI documentation from the output of CLI tools.
+# 
+# If input_file is not found, puts out a warning message in the log and puts a note 
+# in the table that is generated.
+#
+
+from optparse import OptionParser
+import os
+import sys
+
+VERSION = "0.0.2"
+
+LITERAL = '``'
+SPACES = ' ' * 3
+QUOTE = '\''
+BACKSLASH = '\\'
+SECTION_LINE = SPACES + '**'
+COMMAND_LINE = SPACES + LITERAL
+LITERAL_LINE = SPACES + ' | '
+
+SKIP_SECTIONS = ['Security (Beta)']
+
+MISSING_FILE_TEMPLATE = "   **Missing Input File**,\"Missing input file %s\""
+TABLE_HEADER = """.. csv-table::
+   :header: Command,Description
+   :widths: 50, 50
+
+"""
+
+def parse_options():
+    """ Parses args options.
+    """
+    parser = OptionParser(
+        usage="\n\n  %prog input_file output_file",
+        description="Generates an rst table to be included in the documentation")
+
+    parser.add_option(
+        "-v", "--version",
+        action="store_true",
+        dest="version",
+        help="Version of this software",
+        default=False)
+
+    (options, args) = parser.parse_args()
+    
+    if options.version:
+        print "Version: %s" % VERSION
+        sys.exit(1)
+
+    if not args or len(args) != 2:
+        parser.print_help()
+        sys.exit(1)
+
+    return options, args
+
+#
+# Utility functions
+#
+
+def skip_this_section(line):
+    text = line.strip(' *').lower()
+    skip_sections = [section.lower() for section in SKIP_SECTIONS]
+    return text in skip_sections
+    
+def create_literal_line(line):
+    right_line = line.lstrip()
+    spaces = len(line) - len(right_line)
+    return LITERAL_LINE + ' ' * spaces + LITERAL + right_line + LITERAL
+
+def create_parsed_line(line):
+    """ Parses a line and changes single-quotes to a combination of literals and single-quotes.
+        Handles closing literals with a backslash correctly for alpha-numerics.
+        Any text that is enclosed with single-quotes will be converted to a literal with single-quotes.
+        'text' will become ``'text'``
+    """
+    i = 0
+    in_literal = False
+    finished_literal = False
+    opening_literal_quote = LITERAL + QUOTE
+    closing_literal_quote = QUOTE + LITERAL
+    new_line = ''
+    for c in line:
```

Comment:
just wondering - instead of going character by character through the line (and maintaining whether we're in between the quotes or not, and whether we just finished the literal or not), wouldn't a regex replace be simpler?